src/serlo-editor/core/sub-document/editor.tsx

```diff
@@ -59,20 +54,10 @@ export function SubDocumentEditor({ id, pluginProps }: SubDocumentProps) {
 
   useEffect(() => {
     if (domFocusState !== 'focus') return
-    setTimeout(() => autofocusRef.current?.focus())
+    // TODO: does not work any more (with or without timeout)
+    autofocusRef.current?.focus()
```

Comment:
It seems like this never worked 😄 Try deleting the whole effect on staging, autofocus still works.

This is [the line](https://github.com/serlo/frontend/blob/staging/src/serlo-editor/store/focus/slice.ts#L34) that actually sets the focus. On page load, the plugins are added sequentially, but in the reverse order. Each one will be focused in the store, but the first plugin (text plugin of article introduction) will be added last, and therefore focused last. And then this code in SubDocumentEditor makes sure store focus and dom focus are aligned:
```typescript
  const [domFocusState, setDomFocus] = useState<DomFocus>(
    focused ? DomFocus.focusWithin : DomFocus.notFocused
  )
```

It's a hack, but I thought why not try it directly with dom focus. Doesn't work, breaks the editor completely. So I'll explore the possibility of changing the approach. The first idea that I have is to manually focus the article introduction text plugin from the `ArticleEditor` component.

Reply:
Pull request with the implementation of the mentioned idea: https://github.com/serlo/frontend/pull/2780